samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -1239,7 +1247,149 @@ def _link_gateway_method_to_gateway_resource(
+        source_resource_tf_config=gateway_integrations_config_resources,
+        destination_resource_tf=gateway_resources_terraform_resources,
+        tf_destination_attribute_name="id",
+        terraform_link_field_name="resource_id",
+        cfn_link_field_name="ResourceId",
+        terraform_resource_type_prefix=API_GATEWAY_RESOURCE_RESOURCE_ADDRESS_PREFIX,
+        cfn_resource_update_call_back_function=_link_gateway_resource_to_gateway_resource_call_back,
+        linking_exceptions=exceptions,
+    )
+    ResourceLinker(resource_linking_pair).link_resources()
+
+
+def _link_gateway_integration_to_function_call_back(
+    gateway_integration_cfn_resource: Dict, referenced_gateway_resource_values: List[ReferenceType]
+) -> None:
+    """
+    Callback function that is used by the linking algorithm to update an Api Gateway integration CFN Resource with
+    a reference to the Lambda function resource through the AWS_PROXY integration.
+
+    Parameters
+    ----------
+    gateway_integration_cfn_resource: Dict
+        API Gateway integration CFN resource
+    referenced_gateway_resource_values: List[ReferenceType]
+        List of referenced Gateway Resources either as the logical id of Gateway Resource resource
+        defined in the customer project, or ARN values for actual Gateway Resources resource defined
+        in customer's account. This list should always contain one element only.
+    """
+    if len(referenced_gateway_resource_values) > 1:
+        raise InvalidResourceLinkingException(
+            "Could not link multiple Lambda functions to one Gateway integration resource"
+        )
+
+    logical_id = referenced_gateway_resource_values[0]
+    gateway_integration_cfn_resource["Properties"]["Uri"] = (
+        {"Fn::Sub": INVOKE_ARN_FORMAT.format(function_logical_id=logical_id.value)}
+        if isinstance(logical_id, LogicalIdReference)
+        else logical_id.value
+    )
+
+
+def _link_gateway_integrations_to_function_resource(
+    gateway_integrations_config_resources: Dict[str, TFResource],
+    gateway_integrations_config_address_cfn_resources_map: Dict[str, List],
+    lambda_function_terraform_resources: Dict[str, Dict],
+):
+    """
+    Iterate through all the resources and link the corresponding
+    Lambda function resource to each Gateway Integration resource.
+
+    Parameters
+    ----------
+    gateway_integrations_config_resources: Dict[str, TFResource]
+        Dictionary of configuration Gateway Integrations
+    gateway_integrations_config_address_cfn_resources_map: Dict[str, List]
+        Dictionary containing resolved configuration addresses matched up to the cfn Gateway Integration
+    lambda_function_terraform_resources: Dict[str, Dict]
+        Dictionary of all actual terraform Lambda function resources (not configuration resources).
+        The dictionary's key is the calculated logical id for each resource.
+    """
+
```

Comment:
can we filter the dictionary `gateway_integrations_config_resources` for only the resources that has `type` = `AWS_PROXY` , and if there is noting we can skip linking these resources.

Reply:
My point is to avoid throwing limitation exceptions, and it is not required to do this linking.